src/JWT/Builder/JwtBuilder.cs

```diff
@@ -272,18 +278,16 @@ public string Encode(Type payloadType, object payload)
                 throw new ArgumentNullException(nameof(payloadType));
             if (payload is null)
                 throw new ArgumentNullException(nameof(payload));
-            
-            EnsureCanEncode();
 
-            var dic = payloadType.GetProperties(BindingFlags.Instance | BindingFlags.Public)
-                                 .ToDictionary(prop => prop.Name, prop => prop.GetValue(payload, null));
-
-            foreach (var pair in dic)
+            // Preserve old behaviour checking the type, is it needed?
+            if (payloadType != payload.GetType())

```

Comment:
I only added this because once test checked this. The previous implementation needed to know the type to be able to deconstruct the object with reflection.

The new implementation just uses generic serialisation so it's no longer needed to add the type.

Reply:
I'd keep the library lightweight and relying on the underlying serialier to fail